lib/rails/perftest/railties/testing.tasks

```diff
@@ -4,14 +4,25 @@ namespace :test do
   task 'perftest:benchmark_mode' do
     ENV["BENCHMARK_TESTS"] = '1'
   end
+  if Rake::Task.task_defined?('test:prepare')
+    Rails::SubTestTask.new(benchmark: ['test:prepare', 'test:perftest:benchmark_mode']) do |t|
+      t.libs << 'test'
+      t.pattern = 'test/performance/**/*_test.rb'
+    end
 
-  Rails::SubTestTask.new(benchmark: ['test:prepare', 'test:perftest:benchmark_mode']) do |t|
-    t.libs << 'test'
-    t.pattern = 'test/performance/**/*_test.rb'
-  end
+    Rails::SubTestTask.new(profile: 'test:prepare') do |t|
+      t.libs << 'test'
+      t.pattern = 'test/performance/**/*_test.rb'
+    end
+  else
+    Rails::SubTestTask.new(benchmark: ['db:test:prepare', 'test:perftest:benchmark_mode']) do |t|
```

Comment:
Do we really need the `db:test:prepare` dependency? In newer versions of Rails the test database in maintained when running a test. This is ensured by the method [`ActiveRecord::Migration.maintain_test_schema!` in `test_help.rb`](https://github.com/rails/rails/blob/master/railties/lib/rails/test_help.rb#L15). Since `test_help.rb` is required by the generated `test_helper.rb` it's always triggered when running a test.